client/webserver/site/src/js/wallets.ts

```diff
@@ -168,6 +166,28 @@ export default class WalletsPage extends BasePage {
       this.showFiatValue(this.selectedAssetID, amt * conversionFactor, page.sendValue)
     })
 
+    // Clicking on maxSend on the send form should populate the amount field.
+    Doc.bind(page.maxSend, 'click', () => {
+      const asset = app().assets[this.selectedAssetID]
+      if (!asset) return
+      const maxSend = parseFloat(page.maxSend.textContent ?? '0')
+      if ((asset.wallet.traits & traitWithdrawer) === 0) this.populateMaxSend(maxSend * asset.unitInfo.conventional.conversionFactor, asset)
+      else this.populateMaxSend(asset.wallet.balance.available, asset)
+    })
+
+    // Validate send address on input.
+    Doc.bind(page.sendAddr, 'input', async () => {
+      const asset = app().assets[this.selectedAssetID]
+      Doc.hide(page.validAddr)
+      if (!asset || page.sendAddr.value === '') return
+      const addrCheck = {
+        addr: page.sendAddr.value,
+        assetID: asset.id
+      }
+      const res = await postJSON('/api/validateaddress', addrCheck)
+      if (app().checkResponse(res, true)) Doc.show(page.validAddr)
```

Comment:
If `ok: false` in the response (`else`), I'd expect a red X in place of the green check, or perhaps the input's shadow could turn red.

Something like the following is working on my end, although the approach might be wonky:

```diff
diff --git a/client/webserver/site/src/css/wallets.scss b/client/webserver/site/src/css/wallets.scss
index 70d61ffb..05e086a0 100644
--- a/client/webserver/site/src/css/wallets.scss
+++ b/client/webserver/site/src/css/wallets.scss
@@ -56,6 +56,11 @@
     position: absolute;
   }
 
+  #sendAddr.has-error {
+    border-color: #d11414;
+    box-shadow: 0 0 2px 2px #d11414;
+  }
+
   #assetSelect {
     .icon-select {
       padding: 15px;
diff --git a/client/webserver/site/src/js/wallets.ts b/client/webserver/site/src/js/wallets.ts
index 1388bbd8..d392fb9c 100644
--- a/client/webserver/site/src/js/wallets.ts
+++ b/client/webserver/site/src/js/wallets.ts
@@ -179,6 +179,7 @@ export default class WalletsPage extends BasePage {
     Doc.bind(page.sendAddr, 'input', async () => {
       const asset = app().assets[this.selectedAssetID]
       Doc.hide(page.validAddr)
+      page.sendAddr.classList.remove('has-error')
       if (!asset || page.sendAddr.value === '') return
       const addrCheck = {
         addr: page.sendAddr.value,
@@ -186,6 +187,7 @@ export default class WalletsPage extends BasePage {
       }
       const res = await postJSON('/api/validateaddress', addrCheck)
       if (app().checkResponse(res, true)) Doc.show(page.validAddr)
+      else page.sendAddr.classList.add('has-error')
     })
 
     // A link on the wallet reconfiguration form to show/hide the password field.
```